src/main/java/core/basesyntax/service/DataValidator.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dto.FruitTransactionDto;
+
+public interface DataValidator {
+    void validate(FruitTransactionDto fruitTransactionDto);
+
+    void validateFruitType(FruitTransactionDto fruitTransactionDto);
+
+    void validateQuantity(FruitTransactionDto fruitTransactionDto);

```

Comment:
no point in having them public if you call them both in `validate()`, just make them private in the impl class
```suggestion
```